feat(flags): document python UnleashIntegration (#12216)
getsentry/sentry-docs: docs/platforms/python/integrations/unleash/index.mdx

---
title: Unleash
description: "Learn how to use Sentry with Unleash."
---

<PlatformContent includePath="feature-flags/prerelease-alert" />

The [Unleash](https://www.getunleash.io/) integration tracks feature flag evaluations produced by the Unleash SDK. These evaluations are held in memory and sent to Sentry for review and analysis if an error occurs. **At the moment, we only support boolean flag evaluations.**

## Install

> Install `sentry-sdk` (>=TODO:) and `UnleashClient` (>=6.0.1) from PyPI.

Author: TODO after release

```bash
pip install --upgrade sentry-sdk UnleashClient
```

## Configure

Add `UnleashIntegration` to your `integrations` list:

```python
import sentry_sdk
import os

from sentry_sdk.integrations.unleash import UnleashIntegration
from UnleashClient import UnleashClient

unleash_client = UnleashClient(
    url="<Unleash server URL>/api/",  # "http://localhost:4242/api/" if you are self-hosting Unleash
    app_name="my-app",                # Identifies your app in the Unleash UI
    custom_headers={

>         "Authorization": os.environ["UNLEASH_CLIENT_API_TOKEN"]  # See 'client token' docs

Reviewer: This comment could include a url to the docs. 

This code snippet won't only be read while people are on the docs site, people will copy+paste it into their code and then need a way back to the docs pages when the code inevitably needs to change.

    }
)

unleash_integration = UnleashIntegration(unleash_client)

sentry_sdk.init(
    dsn="___PUBLIC_DSN___",
    integrations=[unleash_integration],
)
```

For more information on how to use Unleash, read Unleash's [Python reference](https://docs.getunleash.io/reference/sdks/python) and [quickstart guide](https://docs.getunleash.io/quickstart).

## Verify

> The integration is tested by evaluating a feature flag using your Unleash SDK before capturing an exception.

Reviewer: ```suggestion
Test the integration by evaluating a feature flag using your Unleash SDK before capturing an exception.
```

Author: This would apply to other python FF docs too

```python {tabTitle: Python, using is_enabled}
import sentry_sdk

# Re-use `unleash_client` from the previous step.
test_flag_enabled = unleash_client.is_enabled("test-flag")

sentry_sdk.capture_exception(Exception("Something went wrong!"))
```

```python {tabTitle: Python, using get_variant}
import sentry_sdk

# Re-use `unleash_client` from the previous step.
test_flag_variant = unleash_client.get_variant("test-flag")
test_flag_enabled = test_flag_variant["enabled"]

sentry_sdk.capture_exception(Exception("Something went wrong!"))
```

Visit the Sentry website and confirm that your error event has recorded the feature flag "test-flag", and its value

> is equal to `test_flag_enabled`.

Reviewer: a link into /issues would be good here.

Author: Same as above, this would apply to other python FF docs too

---
title: Unleash
description: "Learn how to use Sentry with Unleash."
---

<PlatformContent includePath="feature-flags/prerelease-alert" />

The [Unleash](https://www.getunleash.io/) integration tracks feature flag evaluations produced by the Unleash SDK. These evaluations are held in memory and sent to Sentry for review and analysis if an error occurs. **At the moment, we only support boolean flag evaluations.**

## Install

Install `sentry-sdk` (>=TODO:) and `UnleashClient` (>=6.0.1) from PyPI.

```bash
pip install --upgrade sentry-sdk UnleashClient
```

## Configure

Add `UnleashIntegration` to your `integrations` list:

```python
import sentry_sdk
import os

from sentry_sdk.integrations.unleash import UnleashIntegration
from UnleashClient import UnleashClient

unleash_client = UnleashClient(
    url="<Unleash server URL>/api/",  # "http://localhost:4242/api/" if you are self-hosting Unleash.
    app_name="my-app",                # Identifies your app in the Unleash UI.
    custom_headers={
        # See https://docs.getunleash.io/how-to/how-to-create-api-tokens
        "Authorization": os.environ["UNLEASH_CLIENT_API_TOKEN"]
    }
)

> unleash_integration = UnleashIntegration(unleash_client)

Reviewer 2: ```suggestion
```

Reviewer 2: I think this is not needed anymore, right?

sentry_sdk.init(
    dsn="___PUBLIC_DSN___",

>     integrations=[unleash_integration],

Reviewer 3: Based on most recent changes to the sdk PR.

```suggestion
    integrations=[UnleashIntegration()],
```
